frostsnapp/lib/backup_workflow.dart

```diff
@@ -0,0 +1,504 @@
+              ),
+            ),
+          ],
+          const Spacer(),
+          Text(
+            'You can pause and resume the backup process at any time.',
+            style: theme.textTheme.bodyMedium?.copyWith(
+              color: theme.colorScheme.onSurfaceVariant,
+            ),
+          ),
+          const SizedBox(height: 8),
+          if (!runOngoing && !lastRunCompleted)
+            Text(
+              'Ready to create your first backup?',
+              style: theme.textTheme.bodyLarge?.copyWith(
+                color: theme.colorScheme.onSurface,
+              ),
+            ),
+          if (!runOngoing && lastRunCompleted)
+            Text(
+              'You have already created backups for this wallet, but you can run through the proccess again if required.',
+              style: theme.textTheme.bodyLarge?.copyWith(
+                color: theme.colorScheme.onSurface,
+              ),
+            ),
+          const SizedBox(height: 16),
+          SizedBox(
+            width: double.infinity,
+            child: FilledButton(
+              onPressed: () {
+                if (!runOngoing && autoStartBackupRun != true) {
+                  manager.maybeStartBackup(accessStructure: accessStructure);
+                }
+                Navigator.push(
+                  context,
+                  MaterialPageRoute(
+                    builder: (context) => BackupChecklistPage(
+                      accessStructure: accessStructure,
+                    ),
+                  ),
+                );
+              },
+              child: Text(runOngoing ? 'Continue' : 'Begin'),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class BackupChecklistPage extends StatefulWidget {
+  final AccessStructure accessStructure;
+  const BackupChecklistPage({super.key, required this.accessStructure});
+
+  @override
+  State<BackupChecklistPage> createState() => _BackupChecklistPageState();
+}
+
+class _BackupChecklistPageState extends State<BackupChecklistPage> {
+  late final Stream<BackupRun> backupStream;
+
+  @override
+  void initState() {
+    super.initState();
+    backupStream = BackupContext.of(context)!
+        .backupStreamForKey(widget.accessStructure.accessStructureRef().keyId);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final settings = SettingsContext.of(context)!;
+    final theme = Theme.of(context);
+    final backgroundColor = ElevationOverlay.applySurfaceTint(
+      theme.colorScheme.surface,
+      theme.colorScheme.surfaceTint,
+      0,
+    );
+
+    return StreamBuilder<BackupRun>(
+      stream: backupStream,
+      builder: (context, snapshot) {
+        final allDevices = widget.accessStructure.devices();
+        final keyId = widget.accessStructure.accessStructureRef().keyId;
+        final manager = BackupContext.of(context)!.manager;
+        final pendingDevices = allDevices
+            .where((deviceId) => !manager.isDeviceComplete(
```

Comment:
What's the point of this `StreamBuilder` when you aren't using the value of snapshot. That's the whole point!

The stream should be a stream of a data struct. Not an opaque type like it is now.